packages/core/test/new-api/reconciliation/futures/reconcileArtifactContractDeployment.ts

```diff
@@ -0,0 +1,222 @@
+
+describe("Reconciliation - artifact contract", () => {
+  const fakeArtifact = ["Fake artifact"] as any;
+
+  const exampleAddress = "0x1F98431c8aD98523631AE4a59f267346ea31F984";
+
+  const exampleDeploymentState: DeploymentExecutionState = {
+    id: "Example",
+    futureType: FutureType.NAMED_CONTRACT_DEPLOYMENT,
+    strategy: "basic",
+    status: ExecutionStatus.STARTED,
+    dependencies: new Set<string>(),
+    history: [],
+    storedArtifactPath: "./artifact.json",
+    storedBuildInfoPath: "./build-info.json",
+    contractName: "Contract1",
+    value: BigInt("0"),
+    constructorArgs: [],
+    libraries: {},
+    from: undefined,
+  };
+
+  it("should reconcile unchanged", () => {
+    const submoduleDefinition = defineModule("Submodule", (m) => {
+      const safeMath = m.library("SafeMath");
+
+      const contract1 = m.contractFromArtifact(
+        "Contract1",
+        fakeArtifact,
+        ["unchanged"],
+        {
+          libraries: {
+            SafeMath: safeMath,
+          },
+        }
+      );
+
+      return { contract1 };
+    });
+
+    const moduleDefinition = defineModule("Module", (m) => {
+      const { contract1 } = m.useModule(submoduleDefinition);
+
+      return { contract1 };
+    });
+
+    assertSuccessReconciliation(moduleDefinition, {
+      "Submodule:SafeMath": {
+        ...exampleDeploymentState,
+        futureType: FutureType.NAMED_LIBRARY_DEPLOYMENT,
+        status: ExecutionStatus.SUCCESS,
+        contractName: "SafeMath",
+        contractAddress: exampleAddress,
+      },
+      "Submodule:Contract1": {
+        ...exampleDeploymentState,
+        futureType: FutureType.ARTIFACT_CONTRACT_DEPLOYMENT,
+        status: ExecutionStatus.STARTED,
+        constructorArgs: ["unchanged"],
+        libraries: {
+          SafeMath: exampleAddress,
+        },
+      },
+    });
+  });
+
+  it("should find changes to contract name unreconciliable", () => {
+    const moduleDefinition = defineModule("Module", (m) => {
+      const contract1 = m.contractFromArtifact(
+        "ContractChanged",
+        fakeArtifact,
+        [],
+        { id: "Example" }
+      );
+
+      return { contract1 };
+    });
+
+    const reconiliationResult = reconcile(moduleDefinition, {
+      "Module:Example": {
+        ...exampleDeploymentState,
+        futureType: FutureType.ARTIFACT_CONTRACT_DEPLOYMENT,
+        status: ExecutionStatus.STARTED,
+        contractName: "ContractUnchanged",
+      },
+    });
+
+    assert.deepStrictEqual(reconiliationResult.reconciliationFailures, [
+      {
+        futureId: "Module:Example",
+        failure:
+          "Contract name has been changed from ContractUnchanged to ContractChanged",
+      },
+    ]);
+  });
+
+  it("should find changes to constructors unreconciliable", () => {
+    const moduleDefinition = defineModule("Module", (m) => {
+      const contract1 = m.contractFromArtifact("Contract1", fakeArtifact, [
+        "changed",
```

Comment:
It would be cool to have a test with more complex arguments, including a future.

Reply:
Same for the positive tests

Reply:
I have added to the tests more complex argument example - including some futures, just for fun.